code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -361,32 +361,34 @@
 		return P
 
 
-/mob/living/simple_animal/hostile/proc/DestroyStructuresCycle(turf/T)
-	var/dir_to_target = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
-	var/list/dir_list = list(dir_to_target, turn(dir_to_target, 90), turn(dir_to_target, 270), turn(dir_to_target, 180))
-	for(var/dir in dir_list)
-		T = get_step(targets_from, dir)
-		for(var/a in T)
-			var/atom/A = a
-			if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
-				A.attack_animal(src)
-				return
+/mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
+	return iswallturf(T) || ismineralturf(T)
 
 
 /mob/living/simple_animal/hostile/proc/DestroyPathToTarget()
-	var/turf/T = get_cardinal_step_towards(targets_from, target)
-		if(environment_smash == ENVIRONMENT_SMASH_RWALLS)
-			if (iswallturf(T) || ismineralturf(T) || isreinforcedwallturf(T))
+	if(environment_smash)
+		EscapeConfinement()
+
+		var/dir_to_target_cardinal = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
```

Comment:
you don't need this at all, since if `if(dir_to_target in GLOB.diagonals)` fails we know the target is either on your turf or in a cardinal direction, so we can add it to the list without checking it.